nion/swift/ExportDialog.py

```diff
@@ -221,56 +223,163 @@ def cancel(self) -> bool:
+        self.height_value_line_edit.text = str(new_height)
+        self.previous_unit_index = new_unit_type
+
+    def convert_to_pixels(self, size: Geometry.FloatSize, unit_type: UnitType) -> Geometry.IntSize:
+        if size.height is None or size.width is None:
+            return Geometry.IntSize(0, 0)
+        else:
+            height = float(size.height)
+            width = float(size.width)
+        if unit_type == UnitType.INCHES:
+            return Geometry.IntSize(height=int(height * PIXELS_PER_INCH), width=int(width * PIXELS_PER_INCH))
+        elif unit_type == UnitType.CENTIMETERS:
+            return Geometry.IntSize(height=int(height * PIXELS_PER_INCH / CM_PER_INCH),
+                                    width=int(width * PIXELS_PER_INCH / CM_PER_INCH))
+        return Geometry.IntSize(height=int(round(height)), width=int(round(width)))  # Pixels
+
+    def convert_from_pixels(self, size: Geometry.IntSize, unit_type: UnitType) -> Geometry.FloatSize:
+        if size.height is None or size.width is None:
+            return Geometry.FloatSize(0.0, 0.0)
+        else:
+            height_px = float(size.height)
+            width_px = float(size.width)
+        if unit_type == UnitType.INCHES:
+            return Geometry.FloatSize(height=height_px / PIXELS_PER_INCH, width=width_px / PIXELS_PER_INCH)
+        elif unit_type == UnitType.CENTIMETERS:
+            return Geometry.FloatSize(height=height_px * CM_PER_INCH / PIXELS_PER_INCH,
+                                      width=width_px * CM_PER_INCH / PIXELS_PER_INCH)
+        return Geometry.FloatSize(width=width_px, height=height_px)  # Pixels
+
     def close(self) -> None:
         pass
 
-
 class ExportSVGDialog:
-
-    def __init__(self, document_controller: DocumentController.DocumentController, display_item: DisplayItem.DisplayItem) -> None:
+    def __init__(self, document_controller: DocumentController.DocumentController,
+                 display_item: DisplayItem.DisplayItem) -> None:
         super().__init__()
-
         self.__document_controller = document_controller
-
         u = Declarative.DeclarativeUI()
-
+        u = Declarative.DeclarativeUI()
         if display_item.display_data_shape and len(display_item.display_data_shape) == 2:
-            display_size = Geometry.IntSize(height=4, width=4)
+            display_size = Geometry.IntSize(height=display_item.display_data_shape[0],
+                                            width=display_item.display_data_shape[1])
         else:
-            display_size = Geometry.IntSize(height=3, width=4)
-
+            display_size = Geometry.IntSize(height=3, width=5)
         handler = ExportSVGHandler(display_size)
 
         def ok_clicked() -> bool:
-            dpi = 96
-            width_px = (handler.width_model.value or display_size.width) * dpi
-            height_px = (handler.height_model.value or display_size.height) * dpi
-
+            display_shape: Geometry.IntSize = handler.convert_to_pixels(
```

Comment:
The `display_shape: Geometry.IntSize. = ` should just be `display_shape = `. The type checker should be able to infer the type from the return value of `handler.convert_to_pixels`.